api/buckets/service.go

```diff
@@ -1606,33 +1607,87 @@ func (s *Service) Archive(ctx context.Context, req *pb.ArchiveRequest) (*pb.Arch
 		return nil, fmt.Errorf("parsing cid path: %s", err)
 	}
 
-	ffsi, err := s.Collections.FFSInstances.Get(ctx, req.GetKey())
-	if err != nil {
-		return nil, fmt.Errorf("getting ffs instance data: %s", err)
+	var ffsInfo *mdb.FFSInfo
+	var owner crypto.PubKey
+	var isAccount bool
+	if acct := accountFromContext(ctx); acct != nil {
+		ffsInfo = acct.FFSInfo
+		owner = acct.Key
+		isAccount = true
+	} else if user := userFromContext(ctx); user != nil {
+		ffsInfo = user.FFSInfo
+		owner = user.Key
+		isAccount = false
+	}
+
+	createNewFFS := func() error {
```

Comment:
Little reusable func that is called to create a new FFS instance and update the `Account` or `User` database record.